packages/botonic-plugin-flow-builder/src/index.ts

```diff
@@ -59,6 +64,22 @@ export default class BotonicPluginFlowBuilder implements Plugin {
       accessToken: this.getAccessToken(request.session),
       request: this.currentRequest,
     })
+
+    if (
+      request.input.data &&
+      !request.input.payload &&
+      !request.session.is_first_interaction
```

Comment:
nit: I would encapsulate this into a function with a descriptive name to make the code easier to read.